src/main/java/com/mraof/minestuck/world/gen/feature/structure/FrogTemplePiece.java

```diff
@@ -0,0 +1,330 @@
+		}
+		if(random.nextBoolean())
+		{
+			int randReduction = random.nextInt(5);
+			fillWithBlocks(world, boundingBox, 18 + 20, -20, 103 + 20, 21 + 20, 40 - randReduction, 108 + 20, block, block, false); //north pillar
+			fillWithBlocks(world, boundingBox, 17 + 20, 41 - randReduction, 102 + 20, 22 + 20, 46 - randReduction, 109 + 20, MSBlocks.BROWN_STONE.getDefaultState(), MSBlocks.CRUXITE_BLOCK.getDefaultState(), false);
+		}
+		if(random.nextBoolean())
+		{
+			int randReduction = random.nextInt(5);
+			fillWithBlocks(world, boundingBox, 67 + 20, -20, 57 + 20, 72 + 20, 40 - randReduction, 60 + 20, block, block, false); //east pillar
+			fillWithBlocks(world, boundingBox, 66 + 20, 41 - randReduction, 56 + 20, 73 + 20, 46 - randReduction, 61 + 20, MSBlocks.BROWN_STONE.getDefaultState(), MSBlocks.URANIUM_BLOCK.getDefaultState(), false);
+		}*/
+	}
+	
+	private void buildFrog(BlockState block, IWorld world, MutableBoundingBox boundingBox)
+	{
+		//fillWithAir(world, boundingBox, 14 + 20, 49, 52 + 20, 27 + 20, 55, 65 + 20); //lotus room
+		//fillWithBlocks(world, boundingBox, 23 + 20, 57, 14 + 38 + 20, 27 + 20, 59, 55 + 20, block, block, false); //right front leg
+		//fillWithBlocks(world, boundingBox, 14 + 20, 57, 14 + 38 + 20, 18 + 20, 59, 55 + 20, block, block, false); //left front leg
+		fillWithBlocks(world, boundingBox, 23 + 20, 57, 14 + 38 + 20, 27 + 20, 58, 55 + 20, block, block, false); //right front leg
+		fillWithBlocks(world, boundingBox, 14 + 20, 57, 14 + 38 + 20, 18 + 20, 58, 55 + 20, block, block, false); //left front leg
+		fillWithBlocks(world, boundingBox, 23 + 20, 57, 23 + 38 + 20, 27 + 20, 61, 64 + 20, block, block, false); //right back leg
+		fillWithBlocks(world, boundingBox, 14 + 20, 57, 23 + 38 + 20, 18 + 20, 61, 64 + 20, block, block, false); //left back leg
+		fillWithBlocks(world, boundingBox, 16 + 20, 57, 16 + 38 + 20, 25 + 20, 67, 63 + 20, block, block, false); //body
+		fillWithBlocks(world, boundingBox, 15 + 20, 65, 14 + 38 + 20, 26 + 20, 72, 59 + 20, block, block, false); //head
+		
+		/*fillWithBlocks(world, boundingBox, 23 + 20, 57, 12 + 38 + 20, 27 + 20, 59, 55 + 20, block, block, false); //right front leg
+		fillWithBlocks(world, boundingBox, 14 + 20, 57, 12 + 38 + 20, 18 + 20, 59, 55 + 20, block, block, false); //left front leg
+		fillWithBlocks(world, boundingBox, 23 + 20, 57, 22 + 38 + 20, 27 + 20, 59, 65 + 20, block, block, false); //right back leg
+		fillWithBlocks(world, boundingBox, 14 + 20, 57, 22 + 38 + 20, 18 + 20, 59, 65 + 20, block, block, false); //left back leg
+		fillWithBlocks(world, boundingBox, 16 + 20, 57, 16 + 38 + 20, 25 + 20, 64, 65 + 20, block, block, false); //body
+		fillWithBlocks(world, boundingBox, 15 + 20, 62, 14 + 38 + 20, 26 + 20, 70, 57 + 20, block, block, false); //head
+		 */
+	}
+	
+	static class Selector extends StructurePiece.BlockSelector {
+		private Selector() {
+		}
+		
+		public void selectBlocks(Random rand, int x, int y, int z, boolean wall)
```

Comment:
That solves it
~~With the means of distribution you chose now, you could clean it up further by defining this through weights instead of indexes (for example with a `WeightedList`)~~
Nevermind, I'm now unsure if that'd be an improvement or not

Reply:
Currently used method is now reliable